node/pkg/db/purge_vaas.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+// This function deletes all VAAs for either the specified chain or specified chain / emitter address
+// that are older than the specified time. If the logOnly flag is specified, it does not delete anything,
+// just counts up what it would have deleted.
+
+func (d *Database) PurgeVaas(prefix VAAID, oldestTime time.Time, logOnly bool) (string, error) {
+	if prefix.Sequence != 0 {
+		return "", fmt.Errorf("may not specify a sequence number on the prefix")
+	}
+
+	numDeleted := 0
+	numKept := 0
+
+	if err := d.db.View(func(txn *badger.Txn) error {
+		it := txn.NewIterator(badger.DefaultIteratorOptions)
+		defer it.Close()
+		prefix := prefix.EmitterPrefixBytes()
+		for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
+			item := it.Item()
+			key := item.Key()
+			err := item.Value(func(val []byte) error {
+				v, err := vaa.Unmarshal(val)
+				if err != nil {
+					return fmt.Errorf("failed to unmarshal VAA for %s: %v", string(key), err)
```

Comment:
Would you ever want to just delete what you can, continue instead of returning on the first error?  Either way there would be some deleted and some not?

Reply:
I think it is okay to just throw an error and give up.